### PR TITLE
Fix unmagic failure for case changes

### PR DESCRIPTION
**Pull request by algmyr**
Just looking at the redirect you don't detect case changes.
Added logic to also check case changes and refactored logic a bit.

### CODE REVIEW

File: tle/cogs/handles.py [hunk 1]
```diff
@@ -437,66 +437,80 @@ async def unmagic(self, ctx):
...
+        # Return summary embed
+        lines = []
+        if not fixed and not failed:
+            return discord_common.embed_success('No handles updated')
+        if fixed:
+            lines.append('**Fixed**')
+            lines += (f'{old} -> {new}' for old, new in fixed)
+        if failed:
+            lines.append('**Failed**')
+            lines += (f'{handle}' for handle in failed)
```
Comment (meooow25) [inline tle/cogs/handles.py:512]: `lines += failed` 👀

File: tle/cogs/handles.py [hunk 2]
```diff
@@ -437,66 +437,80 @@ async def unmagic(self, ctx):
...
+        return discord_common.embed_success('\n'.join(lines))
```
Comment (meooow25) [inline tle/cogs/handles.py:513]: Note: If there are too many and it exceeds char limit, sending will fail 🙃

File: tle/cogs/handles.py [hunk 3]
```diff
@@ -437,66 +437,80 @@ async def unmagic(self, ctx):
...
-                    await cf.user.info(handles=handles)
-                    n_same += len(handles)
-                    break  # no failed handles
+                    users = await cf.user.info(handles=handle_chunk)
+                    lower_map = {h.lower(): h for h in handle_chunk}
+
+                    # Users could still have changed capitalization
+                    def needs_case_fix(user):
+                        return (not user.handle in handle_chunk
+                                and user.handle.lower() in lower_map)
```
Comment (meooow25) [inline tle/cogs/handles.py:478]: > `not user.handle in handle_chunk`

Not sure I understand this.
Also `user.handle.lower()` is expected to be present in `lower_map` even if handle doesn't change, right?

**Review by meooow25 — CHANGES_REQUESTED** [review]

File: tle/cogs/handles.py (hunk 2 above)
Comment (algmyr) [reply tle/cogs/handles.py:513]: I'm aware, I really hope we won't have quite that many users switching handles :|

File: tle/cogs/handles.py (hunk 1 above)
Comment (algmyr) [reply tle/cogs/handles.py:512]: yay for copy paste coding, fixed

File: tle/cogs/handles.py (hunk 3 above)
Comment (algmyr) [reply tle/cogs/handles.py]: Discussed offline and logic updated and simplified.

**Pull request merged**